scopesim/effects/detector_list.py

```diff
@@ -151,48 +163,104 @@ def apply_to(self, obj, **kwargs):
         if not isinstance(obj, FovVolumeList):
             return obj
 
-        hdr = self.image_plane_header
-        xy_mm = calc_footprint(hdr, "D")
-        pixel_size = hdr["CDELT1D"]  # mm
-        pixel_scale = kwargs.get("pixel_scale", self.meta["pixel_scale"])  # ["]
-        pixel_scale = from_currsys(pixel_scale, self.cmds)
+        shrink_axis, shrink_values = self._get_fov_limits(**kwargs)
+        obj.shrink(axis=shrink_axis, values=shrink_values)
+
+        return obj
 
-        # x["] = x[mm] * ["] / [mm]
-        xy_sky = xy_mm * pixel_scale / pixel_size
+    @property
+    def pixel_size(self) -> u.Quantity[u.mm]:
+        """Return size of one pixel in mm."""
+        pixel_size = np.min(
+            quantity_from_table("pixel_size", self.active_table, u.mm))
+        return pixel_size
 
-        obj.shrink(axis=["x", "y"],
-                   values=(tuple(zip(xy_sky.min(axis=0),
-                                     xy_sky.max(axis=0)))))
+    @property
+    def pixel_scale_mm(self) -> u.Equivalency:
+        """Return pixel scale (mm / pix) as equivalency."""
+        return u.pixel_scale(self.pixel_size / u.pixel)
+
+    def _get_pixel_scale_arcsec(self, **kwargs) -> u.Equivalency:
+        """To allow overriding defaults, but use defaults in property."""
+        pixel_scale = u.pixel_scale(
+            from_currsys(
+                kwargs.get("pixel_scale", self.meta["pixel_scale"]),
+                self.cmds,
+            ) << u.arcsec / u.pixel
+        )
+        return pixel_scale
 
-        return obj
+    @property
+    def pixel_scale_arcsec(self) -> u.Equivalency:
+        """Return pixel scale (arcsec / pix) as equivalency."""
+        return self._get_pixel_scale_arcsec()
+
+    def _get_fov_limits(self, **kwargs):
+        sky_points = self._get_corner_points()[:, :2]
+        pixel_scale = self._get_pixel_scale_arcsec(**kwargs)
+
+        with u.set_enabled_equivalencies(pixel_scale + self.pixel_scale_mm):
+            xy_sky = (sky_points << u.pixel).to_value(u.arcsec)
```

Comment:
I don't understand what happens here, could you perhaps add a comment?

The documentation is not at all helpful: https://docs.astropy.org/en/stable/api/astropy.units.set_enabled_equivalencies.html